bokehjs/src/lib/core/util/projections.ts

```diff
@@ -46,17 +46,33 @@ export function in_bounds(value: number, dimension: LatLon): boolean {
   return min < value && value < max
 }
 
-export function project_xy(x: Arrayable<number>, y: Arrayable<number>): [NumberArray, NumberArray] {
+export function $project_xy(x: Arrayable<number>, y: Arrayable<number>, merc_x?: Arrayable<number>, merc_y?: Arrayable<number>): void {
   const n = min(x.length, y.length)
-  const merc_x = new NumberArray(n)
-  const merc_y = new NumberArray(n)
+  merc_x = merc_x ?? x
+  merc_y = merc_x ?? y
   for (let i = 0; i < n; i++) {
     const xi = x[i]
     const yi = y[i]
     const [merc_xi, merc_yi] = wgs84_mercator.compute(xi, yi)
     merc_x[i] = merc_xi
     merc_y[i] = merc_yi
   }
+}
+
+export function $project_xsys(xs: Arrayable<number>[], ys: Arrayable<number>[], merc_xs?: Arrayable<number>[], merc_ys?: Arrayable<number>[]): void {
```

Comment:
Does the `$` signify non-allocating?

Reply:
Yes, though I have mixed feelings about this notation.

Reply:
My preference would be to make those as `/** @internal */` (this has meaning in TS, by excluding such entities from generated declaration files), but that would make them untestable in the current setup. However, I've been experimenting with a different approach to compiling tests, in which internal declarations would be usable, so this may get resolved at some point.

Reply:
I don't have any strong opinion about the notation. Is there a possibility that extension developers might want to be able to use the non-allocating version?